java/code/src/com/redhat/rhn/domain/contentmgmt/ContentProject.java

```diff
@@ -331,7 +331,7 @@ public void attachFilter(ContentFilter filter) {
         if (idx != -1) {
             ContentProjectFilter toUpdate = filters.get(idx);
             if (toUpdate.getState() == ContentProjectFilter.State.DETACHED) {
-                toUpdate.setState(ContentProjectFilter.State.BUILT);
+                toUpdate.setState(ContentProjectFilter.State.EDITED);
```

Comment:
This is not entirely correct. What if the filter wasn't edited at all? OTOH if we wanted to do this properly, we'd need to track the previous state of the object :-/

Reply:
I extracted this logic to a new PR, let's discuss it there.

https://github.com/uyuni-project/uyuni/pull/966